analysis/coverity-scan.sh

```diff
@@ -0,0 +1,105 @@
+#!/bin/bash -eu
+#
+# Copyright 2017 JanusGraph Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+##############################################################################
+#
+# Usage (defaults to https://github.com/JanusGraph/janusgraph project):
+#
+#     $ cd janusgraph
+#     $ env COVERITY_TOKEN="..." COVERITY_EMAIL="..." \
+#           path/to/coverity-scan.sh
+#
+# This script assumes that:
+#
+# * you are scanning a GitHub project
+# * your project is a Java project using Maven
+# * you have a `pom.xml` file at the root of your repo
+# * you are running the build on a Linux 64-bit machine
+#
+# To use for another GitHub repository, run as follows:
+#
+#     $ cd my-repo
+#     $ env COVERITY_TOKEN="..." COVERITY_EMAIL="..." \
+#           GITHUB_ORG="..." GITHUB_PROJECT="..." \
+#           path/to/coverity-scan.sh
+#
+# Note: your Coverity project is assumed to be of the form:
+#
+#     "${GITHUB_ORG}/${GITHUB_PROJECT}"
+#
+# If that is not the case, you can override it as well on the command line via
+# the COVERITY_PROJECT env var.
+#
+# All other settings are either constants in this script or derived from the
+# repository, the `pom.xml` file, or other sources.
+#
+##############################################################################
+
+declare -r COVERITY_TOOL_TAR_GZ="coverity_tool.tar.gz"
+
+# Required to be `cov-int` by Coverity docs:
+# https://scan.coverity.com/download
+declare -r COVERITY_BUILD_OUTPUT_DIR="cov-int"
+declare -r COVERITY_TAR_BZ2="cov-int.tar.bz2"
+
+declare -r COVERITY_TOKEN="${COVERITY_TOKEN:-}"
+declare -r COVERITY_EMAIL="${COVERITY_EMAIL:-}"
+
+if [ -z "${COVERITY_TOKEN}" ]; then
+  echo "Error: env var COVERITY_TOKEN not specified; exiting." >&2
+  exit 1
+elif [ -z "${COVERITY_EMAIL}" ]; then
+  echo "Error: env var COVERITY_EMAIL not specified; exiting." >&2
+  exit 2
+fi
+
+# Get the version number from the `pom.xml` file rather than hardcoding it here; see
+# https://stackoverflow.com/questions/41114695/get-pom-xml-version-with-xmllint
+# for more info. Note that simpler, traditional XPath-based solutions such as
+# https://stackoverflow.com/questions/15461737/how-to-execute-xpath-one-liners-from-shell
+# don't work, because Maven's `pom.xml` files use namespaces.
+declare -r PROJECT_VERSION="$(xmllint --xpath '/*[local-name()="project"]/*[local-name()="version"]/text()' pom.xml)"
+declare -r COVERITY_VERSION="${PROJECT_VERSION}/$(date +'%Y-%m-%d')"
```

Comment:
How about `%Y-%m-%dT%H:%M:%S` to support multiple builds per day? We are technically allowed 3 by project quota, though we haven't needed this so far.